Closes #121 adds MySQL support

kidkool1001 opened the pull request:
adds mysql support via mysql:mysql-connector-java:8.0.33

kidkool1001 committed:
adds MySQL support via
"mysql:mysql-connector-java:8.0.33"

confusedalex commented:
hey @kidkool1001, first thanks for your pr! it's an issue i wanted to work on for a long time, but just didn't had time for it.
currently I see 2 problem with the current implementation:
1. If using SQL, we are still writing and reading directly from the json file, which should be avoided then
2. Spigot already implements a sql database connector, so I don't think we need to add another one

This seems like a good excuse to reorganize the project a little bit and make it more modular. I will work on it

kidkool1001 committed:
- removes "mysql:mysql-connector-java:8.0.33" to use spigot's sql database connector
 - no longer reads/writes to json while in sql mode
 - fixed setAccountBalance to not allow negative numbers

kidkool1001 committed:
Switched to HikariCP for connection pooling to improve performance and scalability.

- Added pool settings to the database config section for server admins to customize
- Updated DatabaseManager to use HikariCP and accept pooling parameters
- Improved config comments for clarity and usability

Connection pooling allows servers with higher populations to handle more concurrent database transactions efficiently, reducing wait times and resource overhead.

kidkool1001 commented:
I will take a look into this in the morning cheers!
```
[06:56:50] [Server thread/ERROR]: [TheGoldEconomy] [STDERR] SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
[06:56:50] [Server thread/WARN]: Nag author(s): '[ConfusedAlex]' of 'TheGoldEconomy v1.9.1' about their usage of System.out/err.print. Please use your plugin's logger instead (JavaPlugin#getLogger).
[06:56:50] [Server thread/ERROR]: [TheGoldEconomy] [STDERR] SLF4J: Defaulting to no-operation (NOP) logger implementation
[06:56:50] [Server thread/ERROR]: [TheGoldEconomy] [STDERR] SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details. 
```

**This is not a critical error, but rather a warning that can be addressed for cleaner logging.**
